src/server/rpc/procedures/baidu-maps/baidu-maps.js

```diff
@@ -0,0 +1,357 @@
+/**
+ * The BaiduMaps Service provides access to the Baidu Maps API to provide an alternative to Google Maps for Chinese users.
+ * For more information, check out http://lbsyun.baidu.com/index.php?title=static
+ *
+ * Terms of use: http://lbsyun.baidu.com/index.php?title=open/question
+ * @service
+ */
+'use strict';
+
+const logger = require('../utils/logger')('baidu-staticmaps');
+const request = require('request');
+const SphericalMercator = require('sphericalmercator');
+const geolib = require('geolib');
+const merc = new SphericalMercator({size:256});
+const CacheManager = require('cache-manager');
+const Storage = require('../../storage');
+
+// TODO: Change this cache to mongo or something (file?)
+// This cache is shared among all BaiduMaps instances
+const cache = CacheManager.caching({store: 'memory', max: 1000, ttl: Infinity});
+const key = process.env.BAIDU_MAPS_KEY;
+
+var storage;
+
+// Retrieving a static map image
+var baseUrl = 'https://api.map.baidu.com/staticimage/v2?mcode=666666&',
+    getStorage = function() {
+        if (!storage) {
+            storage = Storage.create('baidu-staticmaps');
```

Comment:
maybe `baidu-maps` might be better. I am guessing this is a result of copying from the google maps service (which used to be called "staticmaps").

Reply:
Staticmaps is their official term for it, is it worth changing?

Reply:
It's changed now, good point that the service name doesn't match.